streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -663,16 +663,21 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
    */
   final def mapOutZIOPar[Env1 <: Env, OutErr1 >: OutErr, OutElem2](n: Int, bufferSize: Int = 16)(
     f: OutElem => ZIO[Env1, OutErr1, OutElem2]
+  )(implicit trace: Trace): ZChannel[Env1, InErr, InElem, InDone, OutErr1, OutElem2, OutDone] =
+    ZChannel.unwrap(
+      Ref.make(n).map(n0 => mapOutZIOPar[Env1, OutErr1, OutElem2](n0, bufferSize)(f))
+    )
+
+  final def mapOutZIOPar[Env1 <: Env, OutErr1 >: OutErr, OutElem2](n: Ref[Int], bufferSize: Int)(
```

Comment:
not sure how to deal with adding  default value for bufferSize since it's an overloaded method, and scala does not allow multiple overloads with defaults